src/idpLoader.ts

```diff
@@ -0,0 +1,102 @@
+import nodeFetch from "node-fetch";
+import { DOMParser } from "xmldom";
+import { log } from "./utils/logger";
+
+export interface IDPMetadataOptions {
+  cert: ReadonlyArray<string>;
+  entryPoint: string;
+  logoutUrl: string;
+}
+
+interface IDPMetadataParams extends IDPMetadataOptions {
+  entityID: string;
+}
+
+export class IDPMetadata {
+  public cert: ReadonlyArray<string>;
+  public entityID: string;
+  public entryPoint: string;
+  public logoutUrl: string;
+  constructor(params: IDPMetadataParams) {
+    this.cert = params.cert;
+    this.entityID = params.entityID;
+    this.entryPoint = params.entryPoint;
+    this.logoutUrl = params.logoutUrl;
+  }
+
+  public getIDPOption(): IDPMetadataOptions {
+    return {
+      cert: Array.from(this.cert),
+      entryPoint: this.entityID,
+      logoutUrl: this.logoutUrl
+    };
+  }
+}
+
+export const ARUBA_ID = "https://loginspid.aruba.it";
+export const INFOCERT_ID = "https://identity.infocert.it";
+export const INTESA_ID = "https://spid.intesa.it";
+export const LEPIDA_ID = "https://id.lepida.it/idp/shibboleth";
+export const NAMIRIAL_ID = "https://idp.namirialtsp.com/idp";
+export const POSTE_ID = "https://posteid.poste.it";
+export const SIELTE_ID = "https://identity.sieltecloud.it";
+export const SPIDITALIA_ID = "https://spid.register.it";
+export const TIM_ID = "https://login.id.tim.it/affwebservices/public/saml2sso";
+
+const EntityDescriptorTAG = "md:EntityDescriptor";
+const X509CertificateTAG = "ds:X509Certificate";
+const SingleSignOnServiceTAG = "md:SingleSignOnService";
+const SingleLogoutServiceTAG = "md:SingleLogoutService";
+
+export async function parseIdpMetadata(
+  IDP_METADATA_URL: string
+): Promise<ReadonlyArray<IDPMetadata>> {
+  const idpMetadataRequest = await nodeFetch(IDP_METADATA_URL);
+  const ipdMetadataPage = await idpMetadataRequest.text();
+  const domParser = new DOMParser().parseFromString(ipdMetadataPage);
```

Comment:
I suggest to split fetching and parsing in two different functions, makes it easier to test (i.e. you can test parsing by feeding a string)

Reply:
fixed on commit 0bf5c0c